cypher-manual/clauses/call.md

```diff
@@ -0,0 +1,253 @@
+| 'John'     | '123456789' |
+| 'Memgraph' | '555555555' |
++--------------------------+
+```
+
+## 1.4. Observing changes from previous executions
+
+Subqueries can be
+
+```cypher
+UNWIND [0, 1, 2] AS x
+CALL {
+  MATCH (n:Counter)
+    SET n.count = n.count + 1
+  RETURN n.count AS innerCount
+}
+WITH innerCount
+MATCH (n:Counter)
+RETURN
+  innerCount,
+  n.count AS totalCount
+```
+
+Output:
+```nocopy
++------------+-------------+
+| innerCount | totalCount  |
++--------------------------+
+| 1          | 3           |
+| 2          | 3           |
+| 3          | 3           |
++--------------------------+
+```
+
+## 1.5. Unit subqueries
+
+Unit subqueries are used when we want to perform a single action for every node from the input branch.
+Consider the following query, if there was only one `:Person` in the graph. The query allows us to clone the
+person with desired preferences in the `FOREACH` clause.
+
+```cypher
+MATCH (p:Person)
+CALL {
+	FOREACH (i IN range(1, 5) | CREATE(:Person {id: i}))
+};
+
+MATCH (n) RETURN COUNT(n) AS no_created_nodes;
+```
+
+Output:
+```nocopy
++------------------+
+| no_created_nodes |
++------------------+
+| 6                |
++------------------+
+```
+
+# 2. Invalid uses of CALL subquery
+## 2.1. Returning variables with the same name as those in the outer scope
+
+Invalid use:
+```cypher
+MATCH (n:Person)
+CALL {
+  MATCH (n:Parent)
+  RETURN n
+}
+RETURN n;
+```
+
+This query results in a semantic exception, because we already have the variable of same name in the
+outer scope of the variable. By renaming either the outer scope variable, or the subquery return variable,
+we are able to successfully execute the query.
+
+Valid use:
+```cypher
+MATCH (n:Person)
+CALL {
+  MATCH (p:Parent)
+  RETURN p
+}
+RETURN n, p;
+```
+
+## 2.2. Returning non aliased expressions
+
+Invalid use:
+```cypher
+MATCH (n:Person)
+CALL {
+  WITH n
+  MATCH (n)-[:HAS_PARENT]->(parent:Parent)
+  RETURN parent.age
+}
+RETURN n, parent.age;
+```
+
+This query results in a semantic exception since the returned expression has not been aliased and can not be interpreted
+correctly on the exit of the subquery. By aliasing the expression we can reference it correctly on the exit of the subquery
+and use it in the outer scope.
+
+Valid use:
+```cypher
+MATCH (n:Person)
+CALL {
+  WITH n
+  MATCH (n)-[:HAS_PARENT]->(parent:Parent)
+  RETURN parent.age AS parent_age
+}
+RETURN n, parent_age;
+```
+
+## 2.3. Referencing outer scope variables that do not exist
+
+Invalid use:
+```cypher
+MATCH (n:Person)
+CALL {
+  WITH o
+  MATCH (o)-[:HAS_CHILD]->(child:Parent)
+  RETURN child
+}
+RETURN DISTINCT n;
+```
+
+This query results in a semantic exception because the variable from outer scope does not exist.
+We can only execute queries if we referenced the variable that has already been bounded in the input branch going
+into the subquery. By renaming the variable to already bounded variable `n`, we are able to execute the query.
```

Comment:
```suggestion
The above query results in a semantic exception because the variable from the outer scope does not exist.
Queries can be executed only by referencing variables bounded in the input branch to the subquery. By renaming the variable to the already bounded variable `n`, the query will be correctly executed.
```